apps/system/test/marionette/lib/utility_tray.js

```diff
@@ -64,7 +64,13 @@
     },
 
     get quickSettings() {
-      return this.client.helper.waitForElement(this.Selectors.quickSettings);
+      var element;
+      this.client.waitFor(function() {
+        element = this.client.findElement(this.Selectors.quickSettings);
+        return element && element.displayed();
+      }.bind(this));
+
+      return element;
```

Comment:
`waitForElement` basically does the above, but has more robust logic for when an element is not accessible [1]. What advantage does this change have?

1.) https://github.com/mozilla-b2g/gaia/blob/2a6e31cdf1dabad180ff9dbc3eddce440d00ca5e/tests/jsmarionette/plugins/marionette-helper/index.js#L147